### PR TITLE
PROC-1128: Update OSS docs to include json payload type

### PR DESCRIPTION
**Pull request by zacharygoodwin**

### CODE REVIEW

File: docs/specification.md [hunk 1]
```diff
@@ -121,17 +121,18 @@ Arbitrary data can be associated with each test bucket and delivered to your app
...
+| Type          | Java Type            |
+|---------------|----------------------|
+| `stringValue` | `String`             |
+| `stringArray` | `String[]`           |
+| `doubleValue` | `double`             |
+| `doubleArray` | `double[]`           |
+| `longValue`   | `long`               |
+| `longArray`   | `long[]`             |
+| `map`         | `Map<String,Object>` |
+| `json`        | `JsonNode`           |
```
Comment (taitori) [inline docs/specification.md:135]: Shall we include full path like `com.fasterxml.jackson.databind.JsonNode`?

File: docs/using-groups.md [hunk 2]
```diff
@@ -148,6 +148,14 @@ Where ExampleGroupsPayload is another generated class, and to retrieve variable
    // VariableOne and VariableTwo are the names of your variables inside the map payload, capitalized first character 
 </code></pre>
 
+#### json payload returns
+
+json payloads provide an additional method to attempt retrieving the payload as the provided class type inputted as the parameter:
```
Comment (taitori) [inline docs/using-groups.md:153]: I think it's worth mentioning this method throws if the deserialization fails.

Comment (shoichi-indeed) [reply docs/using-groups.md:153]: I think it returns `null` instead of throwing an exception, but sounds it's worth mentioning it anyway.

**Review by shoichi-indeed — APPROVED** [review]
looks good to me

**Pull request merged**